backend/src/api/endpoints/source_document.py

```diff
@@ -284,9 +293,15 @@ async def unlinks_all_tags(
     description="Links a DocumentTag with the SourceDocument with the given ID if it exists",
 )
 async def link_tag(
-    *, db: Session = Depends(get_db_session), sdoc_id: int, tag_id: int
+    *,
+    db: Session = Depends(get_db_session),
+    sdoc_id: int,
+    tag_id: int,
+    authz_user: AuthzUser = Depends(),
 ) -> SourceDocumentRead:
-    # TODO Flo: only if the user has access?
+    authz_user.assert_in_same_project_as(Crud.SOURCE_DOCUMENT, sdoc_id)
+    authz_user.assert_in_same_project_as(Crud.DOCUMENT_TAG, tag_id)
+
```

Comment:
theoretisch kann ich ein sdoc aus projekt 2 mit einem tag aus projekt 3 verknüpfen, oder? Wollen wir sowas eig abfangen?

Reply:
Ja, würde ich schon sagen! In Projekt A dürfen ausschließlich Objekte aus Projekt A verwendet werden dürfen. Selbst, wenn ich auch in Projekt B bin (wo ich dann nur Objekte aus Projekt B verwenden darf).

Denke dafür wäre eine ähnliche Methode gut, also `assert_objects_in_same_project` oder so

Reply:
Das habe ich auch an anderen Stellen schon abgefangen, macht auf jeden Fall sinn das überall zu tun.